code/Model/EditableFormField/EditableFileField.php

```diff
@@ -221,14 +238,22 @@ public function getFormField()
         }
 
         $folder = $this->Folder();
+        
+        if ($protectFolder && $folder->ID === 0) {
```

Comment:
Get rid of `$protectFolder` and move this block into the `if ($folder && $folder->exists()) {` below